FirebaseDatabase/Sources/Api/FIRDatabase.m

```diff
@@ -173,6 +173,17 @@ - (void)purgeOutstandingWrites {
     });
 }
 
+- (void)useEmulatorWithHost:(NSString *)host port:(NSInteger)port {
```

Comment:
(commenting here because I can't comment higher)

Make sure the following sequence works:

1. Get database instance for database `foo.firebaseio.com`
2. Call `useEmulator`
3. Call `refFromUrl('https://foo.firebaseio.com/foo/bar/baz')`

I believe that will currently throw, but it should actually return an emulated ref with path `/foo/bar/baz`

cc @schmidt-sebastian to confirm

Reply:
We discussed this in the Emulator chat. `refFromUrl` should be updated to only throw for production database URLs (those that end in `.firebaseio.com`).

Reply:
After more discussion, we may not validate that the hosts matches at all. Let's wait for Sam to turn this into code before updating this PR.

Reply:
SGTM, I'll hold off on new changes to this PR for now.

Reply:
JS is all merged now if you want to copy that implementation.

Reply:
Done. This flow is now described in `testEmulatedDatabaseValidatesOnlyNonCustomURLs` in `FIRDatabaseTests.m`.